### ISSUE TITLE
uneeded if-before-delete

### ISSUE THREAD

**Issue by ReiquelApplegate** [ZeldaClassic/ZeldaClassic#17]
There are several unnecessary if statements that check if a pointer is non-null before trying to delete it.

**Comment by wilsonpe66** [comment]
people seem to think that delete is the same as free.

in c++ the delete command does a null check and called a destructor if defined.

**Comment by ReiquelApplegate** [comment]
We aren't sure why the previous developers put in those ifs-before-free/delete.  I've decided to take them out so there's less code for us to read, as the codebase size is a big problem today.  If someone thinks it's important to skip calling free/delete somewhere for performance reasons then I am gonna ask that they put a code comment as such in there so it doesn't look like some C++ noob who doesn't know any better and ends up getting refactored away.

**Comment by wilsonpe66** [comment]
> We aren't sure why the previous developers put in those ifs-before-free/delete.  I've decided to take them out so there's less code for us to read, as the codebase size is a big problem today.  If someone thinks it's important to skip calling free/delete somewhere for performance reasons then I am gonna ask that they put a code comment as such in there so it doesn't look like some C++ noob who doesn't know any better and ends up getting refactored away.

Yeah, just be careful about things that are not trivially deletable :) you may very well be adding code in places where you're deleting things just so you can clean them up in the call site instead.
